app/services/data/mock/generate-bill-run.service.js

```diff
@@ -0,0 +1,190 @@
+'use strict'
+
+/**
+ * Generates a mock bill run based on a real one
+ * @module GenerateBillRunService
+ */
+
+const BillingBatchModel = require('../../../models/water/billing-batch.model.js')
+const GenerateMockDataService = require('./generate-mock-data.service.js')
+const MockBillRunPresenter = require('../../../presenters/data/mock-bill-run.presenter.js')
+
+async function go (id) {
+  const billingBatch = await _fetchBillingBatch(id)
+
+  if (!billingBatch) {
+    throw new Error('No matching bill run exists')
+  }
+
+  _mockBillingInvoices(billingBatch.billingInvoices)
+
+  return _response(billingBatch)
+}
+
+async function _fetchBillingBatch (id) {
```

Comment:
Not 100% certain this needs to be `async` seeing as it returns a query, and it's the returned query that we `await`; but not going to quibble over it!